src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,727 @@
+     and in a second round assign x' to x and y' to y
+  *)
+  let assign_var_parallel t vv's =
+    let assigned_vars = List.map (function (v, _) -> v) vv's in
+    let t = add_vars t assigned_vars in
+    let primed_vars = List.init (List.length assigned_vars) (fun i -> Var.of_string (Int.to_string i  ^"'")) in (* TODO: we use primed vars in analysis, conflict? *)
+    let t_primed = add_vars t primed_vars in
+    let multi_t = List.fold_left2 (fun t' v_prime (_,v') -> assign_var t' v_prime v') t_primed primed_vars vv's in
+    match multi_t.d with
+    | Some arr when not @@ is_top multi_t ->
+      let switched_arr = List.fold_left2 (fun multi_t assigned_var primed_var-> assign_var multi_t assigned_var primed_var) multi_t assigned_vars primed_vars in
+      let res = drop_vars switched_arr primed_vars ~del:true in
+      let x = Option.get res.d in
+      {d = Some x; env = res.env}
+    | _ -> t
+
+  let assign_var_parallel t vv's =
+    let res = assign_var_parallel t vv's in
+    if M.tracing then M.tracel "ops" "assign_var parallel: %s -> %s \n" (show t) (show res);
+    res
+
+  let assign_var_parallel t vv's = timing_wrap "var_parallel" (assign_var_parallel t) vv's
+
+  let assign_var_parallel_with t vv's =
+    let t' = assign_var_parallel t vv's in
+    t.d <- t'.d;
+    t.env <- t'.env
+
+  let assign_var_parallel_with t vv's =
+    if M.tracing then M.tracel "var_parallel" "assign_var parallel'\n";
+    assign_var_parallel_with t vv's
+
+  let assign_var_parallel' t vs1 vs2 =
+    let vv's = List.combine vs1 vs2 in
+    assign_var_parallel t vv's
+
+  let assign_var_parallel' t vv's =
+    let res = assign_var_parallel' t vv's in
+    if M.tracing then M.tracel "ops" "assign_var parallel'\n";
+    res
+
+  let substitute_exp t var exp no_ov =
+    let t = if not @@ Environment.mem_var t.env var then add_vars t [var] else t in
+    let res = assign_exp t var exp no_ov in
+    forget_vars res [var]
+
+  let substitute_exp t var exp ov =
+    let res = substitute_exp t var exp ov
+    in if M.tracing then M.tracel "ops" "Substitute_expr t: \n %s \n var: %s \n exp: %a \n -> \n %s\n" (show t) (Var.to_string var) d_exp exp (show res);
+    res
+
+  let substitute_exp t var exp ov = timing_wrap "substitution" (substitute_exp t var exp) ov
+
+  let print_coeff_vec l (env : Environment.t) =
+    let print_element e = match e with
+      | (a, Some x) -> print_string ((Z.to_string a) ^ " * " ^ (Var.to_string ( Environment.var_of_dim env x)) ^ " + ")
+      | (a, None) -> print_string ((Z.to_string a) ^ "+") in
+    List.iter print_element  l; print_newline ()
+
+  let print_final_expr l (env : Environment.t) =
+    let print_element i a = if i = 0 then print_string ((Z.to_string a) ^ " + ") else
+        print_string ((Z.to_string a) ^ " * " ^ (Var.to_string ( Environment.var_of_dim env (i-1))) ^ " + ") in
+    List.iteri print_element l; print_newline ()
```

Comment:
Consider replacing these with side-effect-free functions producing a `string`, in case this is needed.